### ISSUE TITLE
[Rogue][Talent] Prey on the weak

### ISSUE THREAD

**Issue by NikaFK** [dalaranwow/dalaran-wow#3987]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

**Current behaviour**: Tell us what happens in detail (Where, When, What, Why, Who)
it is not applying correctly it is not taking into the account of the current health for the rogue where if the rogue is higher health% it does not apply the 20% crit damage it only applies when the rogue is on 100% health

**Expected behaviour**: Tell us what should happen instead
http://wow.gamepedia.com/Prey_on_the_Weak_(old)
https://db.rising-gods.de/?spell=58670
http://wotlk.openwow.com/spell=58670
it is supposed to scale and track with the rogues current health % where the 20% crit damage is added to all damaging crits the rouge preform

**Steps to reproduce the problem**:

1. damage the dummy when your full health look at average damage of crits
2. damage the dummy after it has lost a few % health then drop a few% but still above the targets health %
3. then do average crit damage tests and see that the talent is not working properly

**Include proofs for this behaviour**
with full health and talent applying with the target less health %
http://imgur.com/a/3Sleq

With 2% less health but still more health % then the dummy but prey on the weak does not apply
http://imgur.com/a/D3kk3

You can research this on several pages like http://wowhead.com or http://wowwiki.wikia.com or http://www.youtube.com/
Issues without proofs will be closed and tag On Hold will be added until the proof is provided.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**
http://wotlk.openwow.com/spell=58670
You can research this on several pages like http://wotlk.openwow.com or http://wowhead.com/

**Include Screenshots from the issue if necessary**

**Comment by Rushor** [comment]
Never use dummies to test spells or talents. Please retest everything with a normal player as victim or a boss/oher creature in the world. Thank you.

Dummies have a different script which makes some spellcalcualtions wrong. Thank you. I will reopen the issue if the issue still exists while fighting against a player.

**Comment by JarlBallin** [comment]
It occours vs players

**Comment by JarlBallin** [comment]
@Rushor

**Comment by Rushor** [comment]
anyone else can confirm this issue?

**Comment by alexuktc** [comment]
After I saw the issue I paid a bit more attention to it. I can confirm it, I saw a substantial decrease on my critical strikes on Skadi while I was not on full HP. I was on higher % than the boss, but my critical strikes were not as high as they should have been.

**Comment by Rushor** [comment]
current code for this seems to be fine:

```c++
void HandleEffectPeriodic(AuraEffect const* /*aurEff*/)
            {
                Unit* target = GetTarget();
                Unit* victim = target->GetVictim();
				if (!victim && target->GetTypeId() == TYPEID_PLAYER)
					victim = target->ToPlayer()->GetSelectedUnit();

                if (victim && (target->GetHealthPct() > victim->GetHealthPct()))
                {
                    if (!target->HasAura(SPELL_ROGUE_PREY_ON_THE_WEAK))
                    {
                        int32 bp = GetSpellInfo()->Effects[EFFECT_0].CalcValue();
                        target->CastCustomSpell(target, SPELL_ROGUE_PREY_ON_THE_WEAK, &bp, 0, 0, true);
                    }
                }
                else
                    target->RemoveAurasDueToSpell(SPELL_ROGUE_PREY_ON_THE_WEAK);
            }
```

sould someone pls test this with 1000 crits against a player?

**Comment by rekless91** [comment]
I have tested this against another player using Gouge since it is static damage.  The talent does not increase damage dealt when my health was higher than the player.

**Comment by ulekki** [comment]
So uhm, after a lot of testing I can confirm that the talent is working with auto attacks, evi and SS. For some reason it does not work with Gouge though. Tested versus another player.

**Comment by Rushor** [comment]
so the issue can be closed then @ulekki ?

**Comment by alexuktc** [comment]
Actually no. I did the tests. Below are the 2 attachments (a third one will be following although I did small testing on that one and it is most certainly bugged). 
1st picture is when the rogue is on full health. The fluctuations in the critical strikes are caused by the fact that at the start of the fight the dummy also had 100% HP, and since we had the same the talent did not proc. 
2nd picture shows the rogue with less health than the dummy. 

For the intents and purposes of this test all proc gear was removed (no trinkets) and I used a mace with 1-3 damage (I did not find the 1-2 daggers, I don't think it's statistically significant). As you can see when you're on higher HP the talent procs. The 20% difference in critical damage is present. 
**However** something struck me while I was doing the second test - at one point the dummie had less HP than me (percentage wise). Yet the talent was not proccing. At least 200 of the auto attacks (and around 80 of the critical strikes) were while the dummie has visibly less HP than me, yet the talent did not proc. 

There is an obvious problem here, as you need to be on full HP for it to register, and that is quite impossible in the current progression fights nor will be possible in ICC. 

![full hp](https://cloud.githubusercontent.com/assets/24550199/22184837/c39e90b6-e0e2-11e6-9856-aabfa8bf7e02.png)

![less than full](https://cloud.githubusercontent.com/assets/24550199/22184839/c8cb32ce-e0e2-11e6-884f-5b9498a02dc4.png)

**Comment by Rushor** [comment]
never test the stuff on dummies xd

**Comment by alexuktc** [comment]
This is a matter of principle. Are you saying that the coding specifically excludes dummies? I don't think it will be different if tested on a player. And unlike the person saying he's done a lot of testing I actually provided recount screenshots. 

I may resort to doing that later, I'll see if I have the time.

**Comment by Edzero** [comment]
> Dummies have a different script which makes some spellcalcualtions wrong. Thank you. I will reopen the issue if the issue still exists while fighting against a player.

**Comment by ulekki** [comment]
I'll just go and take my words back. I'm currently slapping a priest in dala severs. IT SEEEEEEMS that the talent works randomly. Sometimes it does, sometimes it doesnt. Gonna post results later today.

**Comment by ulekki** [comment]
ok I think that Im onto something, looks like the talent is working when you are at 100-99% hp, but stops working when you drop below that. Explains why I saw nothing strange when I was testing this last time, I were always at 99% hp

**Comment by ulekki** [comment]
nvm the shit is even more buggier than that

**Comment by ulekki** [comment]
so yeah sorry, for me the talent is working as long as I keep hitting the target all the time, for example if I stop attacking target for 2 secs, then rupture looses benefit from prey of the weak, same if I stop attacking and evi few secs later then it wont benefit from the talent

**Comment by ulekki** [comment]
thats vs another player

**Comment by Rushor** [comment]
Closing and setting this "On Hold" for now, because i could not find any way to reproduce this on PTR.

Once a tester found a way to reproduce it, the ticket will be reopened. Thanks.

**Comment by Jiaku** [comment]
The only way to reproduce the issue is if your HP is not 100% or 99% then you are loosing benefits from talent or if you have to switch to another target(add let's say) when you come back to your previous target(boss let's say) you are loosing benefits from talent.
